commands/command_pre_push.go

```diff
@@ -77,14 +78,37 @@ func prePushCommand(cmd *cobra.Command, args []string) {
 			continue
 		}
 
-		pointerCh, err := gitscanner.ScanLeftToRemote(left)
+		pointers, err := scanLeftOrAll(gitscanner, left)
 		if err != nil {
-			Panic(err, "Error scanning for Git LFS files")
+			Print("Error scanning for Git LFS files in %q", left)
+			ExitWithError(err)
 		}
-		upload(ctx, pointerCh)
+		uploadPointers(ctx, pointers)
 	}
 }
 
+func scanLeft(g *lfs.GitScanner, ref string) ([]*lfs.WrappedPointer, error) {
+	var pointers []*lfs.WrappedPointer
+	var multiErr error
+	cb := func(p *lfs.WrappedPointer, err error) {
+		if err != nil {
+			if multiErr != nil {
+				multiErr = fmt.Errorf("%v\n%v", multiErr, err)
+			} else {
+				multiErr = err
+			}
+			return
+		}
+
+		pointers = append(pointers, p)
+	}
+
+	if err := g.ScanLeftToRemote(ref, cb); err != nil {
```

Comment:
Yes. If this returns an error, the command that `ScanLeftToRemote()` attempts to spin up errored, so the callback potentially sets `multiErr` is never called.